### PR TITLE
Refactor patch for optimize_webui.py build script to use brave-specific file

### PR DESCRIPTION
**Pull request by petemill**
There are additions coming to this file to route more webui pages (such as for the `chrome://history` page). This refactor allows those additions to modify the brave version of this script instead of further modifying the patch to the chromium version of the script.

Fix https://github.com/brave/brave-browser/issues/5233

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- `chrome://settings` still works and uses brave fonts

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/resources/brave_optimize_webui_config.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    # Exclude anything that contains dynamic content, e.g. l18n strings
+    # or is large and can be cached across webui pages.
+    '--exclude', 'chrome://brave-resources/fonts/muli.css',
+    '--exclude', 'chrome://brave-resources/fonts/poppins.css',
+  ]
+
+def url_mappings(src_path):
+  # Provide mappings from URL paths to FS locations
+  br_resources_path = os.path.join(src_path, 'brave', 'ui', 'webui', 'resources')
+  br_history_resources_path = os.path.join(src_path, 'brave', 'browser', 'resources', 'history')
```
Comment (mkarolin) [inline browser/resources/brave_optimize_webui_config.py]: Is this just a placeholder for the future?

Comment (petemill) [reply browser/resources/brave_optimize_webui_config.py]: Thanks for noticing - accidentally left that in as I'm making this PR separated out from an actual feature branch

Comment (petemill) [reply browser/resources/brave_optimize_webui_config.py]: Removed now

**Comment by petemill** [comment]
@mkarolin thanks for approval - I added an addition to make sure that dependent builds re-build when this new brave-specific file is modified. I'd appreciate a re-review when you get a chance  :-)

**Comment by rossmoody** [comment]
Is there any reason why this PR might make it so you can't clear your history? I built this one working on my little icon side project and I can't seem to get the clear browsing data modal to appear in settings.

**Comment by mkarolin** [comment]
@rossmoody Clear Browsing Data dialog got broken by the c76 bump. The fix is pending approval here: https://github.com/brave/brave-core/pull/2903/commits/4f8e2b27f553dfad670f2cd42b0653c7f752359a

**Comment by bsclifton** [comment]
@petemill is this one we still want to do? Or should we close the issue / this PR?

**Comment by bsclifton** [comment]
Closing as stale

**Pull request closed without merging**